### DISCUSSION TITLE
Adding solutions to vlm_systems

### DISCUSSION THREAD

**Discussion by juddmehr** [byuflowlab/FLOWUnsteady#31] [Q&A] [answered]
I'm trying to use the `_addsolution()` function from FLOWVLM to add extra outputs that I can view in the saved vtk files.  I've looked at examples elsewhere in FLOWUnsteady, and I'm fairly confident that I've recreated a case that should work, but I'm finding that the solutions I think I'm adding aren't actually making it to the vtk files.  For example, I have the following line, where Vz has the same type and size as  examples I've found, and the rest of the code is identical to the examples, but there is no Vz field in any of the vtk files that are being safed.

`vlm._addsolution(vlm_system, "Vz", Vz, t=PFIELD.t)`

Perhaps I'm missing some flag or initialization for my custom fields?

**Comment by EdoAlvarezR** [comment, marked as answer]
You are doing things correctly, however, the issue is in the architecture of FLOWVLM.

**THE ISSUE**
In order to catch the potential bug of having the user request FLOWVLM to calculate a field with incomplete information (e.g., request calculating aerodynamic forces without an existing circulation solution), I hardcoded the data structures that are allowed as solution field through the module-wide variable `FIELDS`  in FLOWVLM, defined in Line 43 here [LINK](https://github.com/byuflowlab/FLOWVLM/blob/15aa47aa0551e333e54aa5a23cc8bb2231159142/src/FLOWVLM.jl). This dictionary states the necessary dependencies (for instance, `CL` requires an existing `Gamma` solution field in Line 53), and also whether the solution field is a scalar or vector field. Because of the need of knowing whether it's a scalar or a vector, **the function that generates the VTKs (`vlm.save(...)`) checks if the solution field is in `FIELDS` and if it is not there, then it ignores it when writing the VTK**.

**THE SOLUTION**
In order to have FLOWVLM output your `Vz` field, you'll have to also declare its data structure in `FLOWVLM.FIELDS` specifying whether it's a scalar or a vector as follows:

```julia
vlm.FIELDS["Vz"] = [[], "scalar"]        # NOTE: The empty array means that it has no dependencies
vlm._addsolution(vlm_system, "Vz", Vz, t=PFIELD.t)
```

That should do the trick.
